Mathlib/LinearAlgebra/Matrix/SVD/KernelConjTransposeMul.lean

```diff
@@ -0,0 +1,48 @@
+/-
+Copyright (c) 2023 Mohanad ahmed. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mohanad Ahmed
+-/
+
+import Mathlib.LinearAlgebra.Matrix.Hermitian
+
+/-! # Kernel of the products of a Matrix Conjugate Transpose of a Matrix -/
+open Matrix BigOperators
+
+namespace Matrix
+variable {𝕂: Type}[IsROrC 𝕂][DecidableEq 𝕂]
+variable {m n p : Type}
+variable [Fintype m][DecidableEq m][Fintype n]
+variable [DecidableEq n][Fintype l][DecidableEq l]
+
+lemma conjTranspose_mul_self_eq_zero_iff (A: Matrix m n 𝕂):
```

Comment:
Does this generalize to `StarOrderedRing`?

Reply:
Yes. See #6587